common-rest/src/main/java/org/uniprot/api/rest/validation/ValidDownloadRequest.java

```diff
@@ -0,0 +1,121 @@
+package org.uniprot.api.rest.validation;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.uniprot.api.rest.output.UniProtMediaType.*;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorContextImpl;
+import org.springframework.http.MediaType;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+import org.uniprot.api.rest.output.UniProtMediaType;
+import org.uniprot.api.rest.request.DownloadRequest;
+import org.uniprot.core.util.Utils;
+
+@Constraint(validatedBy = ValidDownloadRequest.DownloadRequestValidator.class)
+@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidDownloadRequest {
+
+    /** formats where 'fields' param doesn't make sense and hence not allowed */
+    List<String> FORMATS_WITH_NO_PROJECTION =
+            List.of(
+                    FF_MEDIA_TYPE_VALUE,
+                    LIST_MEDIA_TYPE_VALUE,
+                    MediaType.APPLICATION_XML_VALUE,
```

Comment:
I would prefer this to be supplied as a static import as well. Just for appearance